File: AN_2/SEM_2/MPP/java/lab3/src/main/java/ro/mpp2024/domain/ParticipantCursa.java
package ro.mpp2024.domain;

import java.util.Objects;

public class ParticipantCursa{
    private Participant part;
    private Cursa cursa;

    public ParticipantCursa(Participant part, Cursa cursa) {
        this.part = part;
        this.cursa = cursa;
    }

    public Participant getParticipant() {
        return part;
    }

    public void setParticipant(Participant part) {
        this.part = part;
    }

    public Cursa getCursa() {
        return cursa;
    }

    public void setCursa(Cursa cursa) {
        this.cursa = cursa;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        ParticipantCursa that = (ParticipantCursa) o;
        return Objects.equals(part, that.part) && Objects.equals(cursa, that.cursa);
    }

    @Override
    public int hashCode() {
        return Objects.hash(part, cursa);
    }

    @Override
    public String toString() {
        return "ParticipantCursa{" +
                "part=" + part +
                ", cursa=" + cursa +
                '}';
    }
}

File: AN_2/SEM_2/MPP/c#/lab2_1/domain/ParticipantCursa.cs
namespace lab2_1.domain;

public class ParticipantCursa
{
    public Participant part{ get; set; }
    public Cursa cursa { get; set; }

    public ParticipantCursa(Participant part, Cursa cursa)
    {
        part = part;
        cursa = cursa;
    }

    public override string ToString()
    {
        return $"Participant {part} inscris la cursa {cursa}";
    }
}